{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"test.ipynb","provenance":[],"collapsed_sections":[],"mount_file_id":"109unadm6VVX7NFBJUOX5GflcDug-7ASE","authorship_tag":"ABX9TyMwEl+ddWeCmjDdE0x4079h"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":null,"metadata":{"id":"J-yEGKqvsOeR"},"outputs":[],"source":["#load img -> face dettection annd extract preprocess_input\n","#find the cosine distance of current image with all the 8655 features\n","#recommend that image"]},{"cell_type":"code","source":["\n","import numpy as np\n","import pickle\n","from sklearn.metrics.pairwise import cosine_similarity\n","import cv2\n","\n","from PIL import Image"],"metadata":{"id":"8BvU4AL3smu-","executionInfo":{"status":"ok","timestamp":1650193795777,"user_tz":-330,"elapsed":1124,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}}},"execution_count":1,"outputs":[]},{"cell_type":"code","source":["!pip install keras_applications==1.0.8"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"pXl8hswCiGIO","executionInfo":{"status":"ok","timestamp":1650193823280,"user_tz":-330,"elapsed":3802,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}},"outputId":"eecf0250-eed0-4613-e5b1-6bc8703c4ae5"},"execution_count":6,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting keras_applications==1.0.8\n","  Downloading Keras_Applications-1.0.8-py3-none-any.whl (50 kB)\n","\u001b[?25l\r\u001b[K     |██████▌                         | 10 kB 21.7 MB/s eta 0:00:01\r\u001b[K     |█████████████                   | 20 kB 11.9 MB/s eta 0:00:01\r\u001b[K     |███████████████████▍            | 30 kB 9.0 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████▉      | 40 kB 3.6 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████████| 50 kB 2.3 MB/s \n","\u001b[?25hRequirement already satisfied: numpy>=1.9.1 in /usr/local/lib/python3.7/dist-packages (from keras_applications==1.0.8) (1.21.5)\n","Requirement already satisfied: h5py in /usr/local/lib/python3.7/dist-packages (from keras_applications==1.0.8) (3.1.0)\n","Requirement already satisfied: cached-property in /usr/local/lib/python3.7/dist-packages (from h5py->keras_applications==1.0.8) (1.5.2)\n","Installing collected packages: keras-applications\n","Successfully installed keras-applications-1.0.8\n"]}]},{"cell_type":"code","source":["feature_list = np.array(pickle.load(open('/content/drive/MyDrive/celebrity/embedding.pkl','rb')))\n","filenames = pickle.load(open('/content/drive/MyDrive/celebrity/filenames.pkl','rb'))\n"],"metadata":{"id":"sBN2195CdTVs","executionInfo":{"status":"ok","timestamp":1650193818218,"user_tz":-330,"elapsed":3099,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}}},"execution_count":5,"outputs":[]},{"cell_type":"code","source":["!pip install keras_vggface"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"910Q6kGpX7Bn","executionInfo":{"status":"ok","timestamp":1650193815128,"user_tz":-330,"elapsed":4202,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}},"outputId":"d4238f53-fbed-4ee9-fd0d-b5d1700994dd"},"execution_count":4,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting keras_vggface\n","  Downloading keras_vggface-0.6-py3-none-any.whl (8.3 kB)\n","Requirement already satisfied: pillow in /usr/local/lib/python3.7/dist-packages (from keras_vggface) (7.1.2)\n","Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.7/dist-packages (from keras_vggface) (1.15.0)\n","Requirement already satisfied: keras in /usr/local/lib/python3.7/dist-packages (from keras_vggface) (2.8.0)\n","Requirement already satisfied: scipy>=0.14 in /usr/local/lib/python3.7/dist-packages (from keras_vggface) (1.4.1)\n","Requirement already satisfied: pyyaml in /usr/local/lib/python3.7/dist-packages (from keras_vggface) (3.13)\n","Requirement already satisfied: numpy>=1.9.1 in /usr/local/lib/python3.7/dist-packages (from keras_vggface) (1.21.5)\n","Requirement already satisfied: h5py in /usr/local/lib/python3.7/dist-packages (from keras_vggface) (3.1.0)\n","Requirement already satisfied: cached-property in /usr/local/lib/python3.7/dist-packages (from h5py->keras_vggface) (1.5.2)\n","Installing collected packages: keras-vggface\n","Successfully installed keras-vggface-0.6\n"]}]},{"cell_type":"code","source":["!pip install mtcnn"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"10YmaWCqZNIh","executionInfo":{"status":"ok","timestamp":1650193810935,"user_tz":-330,"elapsed":8477,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}},"outputId":"56bad976-94ff-4cb9-9868-e0eabe3f6ccb"},"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting mtcnn\n","  Downloading mtcnn-0.1.1-py3-none-any.whl (2.3 MB)\n","\u001b[K     |████████████████████████████████| 2.3 MB 4.2 MB/s \n","\u001b[?25hRequirement already satisfied: opencv-python>=4.1.0 in /usr/local/lib/python3.7/dist-packages (from mtcnn) (4.1.2.30)\n","Requirement already satisfied: keras>=2.0.0 in /usr/local/lib/python3.7/dist-packages (from mtcnn) (2.8.0)\n","Requirement already satisfied: numpy>=1.14.5 in /usr/local/lib/python3.7/dist-packages (from opencv-python>=4.1.0->mtcnn) (1.21.5)\n","Installing collected packages: mtcnn\n","Successfully installed mtcnn-0.1.1\n"]}]},{"cell_type":"code","source":["from keras_vggface.utils import preprocess_input\n","import tensorflow.python.keras.engine\n","from keras_vggface.vggface import VGGFace\n","from mtcnn import MTCNN\n","\n","\n","\n","model = VGGFace(model='resnet50',include_top=False,input_shape=(224,224,3),pooling='avg')\n","\n","detector = MTCNN()\n","# load img -> face detection\n","sample_img = cv2.imread('/content/drive/MyDrive/celebrity/sample/kat.jpg')\n","results = detector.detect_faces(sample_img)\n","\n","x,y,width,height = results[0]['box']\n","\n","face = sample_img[y:y+height,x:x+width]\n","\n","#  extract its features\n","image = Image.fromarray(face)\n","image = image.resize((224,224))\n","\n","face_array = np.asarray(image)\n","\n","face_array = face_array.astype('float32')\n","\n","expanded_img = np.expand_dims(face_array,axis=0)\n","preprocessed_img = preprocess_input(expanded_img)\n","result = model.predict(preprocessed_img).flatten()\n","#print(result)\n","#print(result.shape)\n","# find the cosine distance of current image with all the 8655 features\n","similarity = []\n","for i in range(len(feature_list)):\n","    similarity.append(cosine_similarity(result.reshape(1,-1),feature_list[i].reshape(1,-1))[0][0])\n","\n","index_pos = sorted(list(enumerate(similarity)),reverse=True,key=lambda x:x[1])[0][0]\n","\n","temp_img = cv2.imread(filenames[index_pos])\n","cv2.waitKey(0)\n"],"metadata":{"id":"aG3xAbzQeEZk","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1650194134752,"user_tz":-330,"elapsed":21530,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}},"outputId":"6e76c7c6-2341-410b-e7b3-a5c4bba6382c"},"execution_count":9,"outputs":[{"output_type":"stream","name":"stdout","text":["Downloading data from https://github.com/rcmalli/keras-vggface/releases/download/v2.0/rcmalli_vggface_tf_notop_resnet50.h5\n","94699520/94694792 [==============================] - 8s 0us/step\n","WARNING:tensorflow:5 out of the last 10 calls to <function Model.make_predict_function.<locals>.predict_function at 0x7f2cbebf5e60> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has experimental_relax_shapes=True option that relaxes argument shapes that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/tutorials/customization/performance#python_or_tensor_args and https://www.tensorflow.org/api_docs/python/tf/function for  more details.\n","WARNING:tensorflow:6 out of the last 11 calls to <function Model.make_predict_function.<locals>.predict_function at 0x7f2cbebcc8c0> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has experimental_relax_shapes=True option that relaxes argument shapes that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/tutorials/customization/performance#python_or_tensor_args and https://www.tensorflow.org/api_docs/python/tf/function for  more details.\n"]},{"output_type":"execute_result","data":{"text/plain":["-1"]},"metadata":{},"execution_count":9}]},{"cell_type":"code","source":["!pip install keras==2.4.3"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Y_CjlolhZLog","executionInfo":{"status":"ok","timestamp":1650193898978,"user_tz":-330,"elapsed":5590,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}},"outputId":"368decd9-eeba-4026-ec18-2f3ced279631"},"execution_count":8,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting keras==2.4.3\n","  Downloading Keras-2.4.3-py2.py3-none-any.whl (36 kB)\n","Requirement already satisfied: h5py in /usr/local/lib/python3.7/dist-packages (from keras==2.4.3) (2.10.0)\n","Requirement already satisfied: numpy>=1.9.1 in /usr/local/lib/python3.7/dist-packages (from keras==2.4.3) (1.18.5)\n","Requirement already satisfied: scipy>=0.14 in /usr/local/lib/python3.7/dist-packages (from keras==2.4.3) (1.4.1)\n","Requirement already satisfied: pyyaml in /usr/local/lib/python3.7/dist-packages (from keras==2.4.3) (3.13)\n","Requirement already satisfied: six in /usr/local/lib/python3.7/dist-packages (from h5py->keras==2.4.3) (1.15.0)\n","Installing collected packages: keras\n","  Attempting uninstall: keras\n","    Found existing installation: keras 2.8.0\n","    Uninstalling keras-2.8.0:\n","      Successfully uninstalled keras-2.8.0\n","Successfully installed keras-2.4.3\n"]}]},{"cell_type":"code","source":["!pip install tensorflow==2.3.1"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":1000},"id":"cHYr8MdYc68U","executionInfo":{"status":"ok","timestamp":1650193893397,"user_tz":-330,"elapsed":63568,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}},"outputId":"8339eb23-f6b6-4c27-b809-3d9ee6219a69"},"execution_count":7,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting tensorflow==2.3.1\n","  Downloading tensorflow-2.3.1-cp37-cp37m-manylinux2010_x86_64.whl (320.4 MB)\n","\u001b[K     |████████████████████████████████| 320.4 MB 26 kB/s \n","\u001b[?25hCollecting tensorflow-estimator<2.4.0,>=2.3.0\n","  Downloading tensorflow_estimator-2.3.0-py2.py3-none-any.whl (459 kB)\n","\u001b[K     |████████████████████████████████| 459 kB 54.5 MB/s \n","\u001b[?25hRequirement already satisfied: absl-py>=0.7.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (1.0.0)\n","Requirement already satisfied: wheel>=0.26 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (0.37.1)\n","Requirement already satisfied: google-pasta>=0.1.8 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (0.2.0)\n","Requirement already satisfied: wrapt>=1.11.1 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (1.14.0)\n","Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (1.1.0)\n","Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (3.3.0)\n","Requirement already satisfied: tensorboard<3,>=2.3.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (2.8.0)\n","Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (1.15.0)\n","Requirement already satisfied: protobuf>=3.9.2 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (3.17.3)\n","Requirement already satisfied: grpcio>=1.8.6 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (1.44.0)\n","Collecting numpy<1.19.0,>=1.16.0\n","  Downloading numpy-1.18.5-cp37-cp37m-manylinux1_x86_64.whl (20.1 MB)\n","\u001b[K     |████████████████████████████████| 20.1 MB 12.0 MB/s \n","\u001b[?25hCollecting h5py<2.11.0,>=2.10.0\n","  Downloading h5py-2.10.0-cp37-cp37m-manylinux1_x86_64.whl (2.9 MB)\n","\u001b[K     |████████████████████████████████| 2.9 MB 19.7 MB/s \n","\u001b[?25hRequirement already satisfied: astunparse==1.6.3 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (1.6.3)\n","Collecting gast==0.3.3\n","  Downloading gast-0.3.3-py2.py3-none-any.whl (9.7 kB)\n","Requirement already satisfied: keras-preprocessing<1.2,>=1.1.1 in /usr/local/lib/python3.7/dist-packages (from tensorflow==2.3.1) (1.1.2)\n","Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /usr/local/lib/python3.7/dist-packages (from tensorboard<3,>=2.3.0->tensorflow==2.3.1) (0.4.6)\n","Requirement already satisfied: setuptools>=41.0.0 in /usr/local/lib/python3.7/dist-packages (from tensorboard<3,>=2.3.0->tensorflow==2.3.1) (57.4.0)\n","Requirement already satisfied: werkzeug>=0.11.15 in /usr/local/lib/python3.7/dist-packages (from tensorboard<3,>=2.3.0->tensorflow==2.3.1) (1.0.1)\n","Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.7/dist-packages (from tensorboard<3,>=2.3.0->tensorflow==2.3.1) (3.3.6)\n","Requirement already satisfied: google-auth<3,>=1.6.3 in /usr/local/lib/python3.7/dist-packages (from tensorboard<3,>=2.3.0->tensorflow==2.3.1) (1.35.0)\n","Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in /usr/local/lib/python3.7/dist-packages (from tensorboard<3,>=2.3.0->tensorflow==2.3.1) (0.6.1)\n","Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /usr/local/lib/python3.7/dist-packages (from tensorboard<3,>=2.3.0->tensorflow==2.3.1) (1.8.1)\n","Requirement already satisfied: requests<3,>=2.21.0 in /usr/local/lib/python3.7/dist-packages (from tensorboard<3,>=2.3.0->tensorflow==2.3.1) (2.23.0)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.7/dist-packages (from google-auth<3,>=1.6.3->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (0.2.8)\n","Requirement already satisfied: cachetools<5.0,>=2.0.0 in /usr/local/lib/python3.7/dist-packages (from google-auth<3,>=1.6.3->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (4.2.4)\n","Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.7/dist-packages (from google-auth<3,>=1.6.3->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (4.8)\n","Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/local/lib/python3.7/dist-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (1.3.1)\n","Requirement already satisfied: importlib-metadata>=4.4 in /usr/local/lib/python3.7/dist-packages (from markdown>=2.6.8->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (4.11.3)\n","Requirement already satisfied: typing-extensions>=3.6.4 in /usr/local/lib/python3.7/dist-packages (from importlib-metadata>=4.4->markdown>=2.6.8->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (4.1.1)\n","Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/dist-packages (from importlib-metadata>=4.4->markdown>=2.6.8->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (3.8.0)\n","Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /usr/local/lib/python3.7/dist-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (0.4.8)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests<3,>=2.21.0->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (2021.10.8)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests<3,>=2.21.0->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (1.24.3)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests<3,>=2.21.0->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (3.0.4)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests<3,>=2.21.0->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (2.10)\n","Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.7/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard<3,>=2.3.0->tensorflow==2.3.1) (3.2.0)\n","Installing collected packages: numpy, tensorflow-estimator, h5py, gast, tensorflow\n","  Attempting uninstall: numpy\n","    Found existing installation: numpy 1.21.5\n","    Uninstalling numpy-1.21.5:\n","      Successfully uninstalled numpy-1.21.5\n","  Attempting uninstall: tensorflow-estimator\n","    Found existing installation: tensorflow-estimator 2.8.0\n","    Uninstalling tensorflow-estimator-2.8.0:\n","      Successfully uninstalled tensorflow-estimator-2.8.0\n","  Attempting uninstall: h5py\n","    Found existing installation: h5py 3.1.0\n","    Uninstalling h5py-3.1.0:\n","      Successfully uninstalled h5py-3.1.0\n","  Attempting uninstall: gast\n","    Found existing installation: gast 0.5.3\n","    Uninstalling gast-0.5.3:\n","      Successfully uninstalled gast-0.5.3\n","  Attempting uninstall: tensorflow\n","    Found existing installation: tensorflow 2.8.0\n","    Uninstalling tensorflow-2.8.0:\n","      Successfully uninstalled tensorflow-2.8.0\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","tables 3.7.0 requires numpy>=1.19.0, but you have numpy 1.18.5 which is incompatible.\n","jaxlib 0.3.2+cuda11.cudnn805 requires numpy>=1.19, but you have numpy 1.18.5 which is incompatible.\n","jax 0.3.4 requires numpy>=1.19, but you have numpy 1.18.5 which is incompatible.\n","datascience 0.10.6 requires folium==0.2.1, but you have folium 0.8.3 which is incompatible.\n","albumentations 0.1.12 requires imgaug<0.2.7,>=0.2.5, but you have imgaug 0.2.9 which is incompatible.\u001b[0m\n","Successfully installed gast-0.3.3 h5py-2.10.0 numpy-1.18.5 tensorflow-2.3.1 tensorflow-estimator-2.3.0\n"]},{"output_type":"display_data","data":{"application/vnd.colab-display-data+json":{"pip_warning":{"packages":["numpy"]}}},"metadata":{}}]},{"cell_type":"code","source":["!pip install MTCNN==0.1.0"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"GjCH7bQfdPey","executionInfo":{"status":"ok","timestamp":1649879188351,"user_tz":-330,"elapsed":2547,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}},"outputId":"4c4de6ec-89cf-4898-be5d-68f42b5e8759"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: MTCNN==0.1.0 in /usr/local/lib/python3.7/dist-packages (0.1.0)\n","Requirement already satisfied: opencv-python>=4.1.0 in /usr/local/lib/python3.7/dist-packages (from MTCNN==0.1.0) (4.1.2.30)\n","Requirement already satisfied: keras>=2.0.0 in /usr/local/lib/python3.7/dist-packages (from MTCNN==0.1.0) (2.8.0)\n","Requirement already satisfied: numpy>=1.14.5 in /usr/local/lib/python3.7/dist-packages (from opencv-python>=4.1.0->MTCNN==0.1.0) (1.21.5)\n"]}]},{"cell_type":"code","source":["!pip install keras_vggface==0.6"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"_I7rFxPqcqv-","executionInfo":{"status":"ok","timestamp":1649879196287,"user_tz":-330,"elapsed":2968,"user":{"displayName":"Akanshu Rana","userId":"05771508798819086616"}},"outputId":"92cbeeab-ac1f-45cf-f999-cbad55282e41"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting keras_vggface==0.6\n","  Using cached keras_vggface-0.6-py3-none-any.whl (8.3 kB)\n","Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.7/dist-packages (from keras_vggface==0.6) (1.15.0)\n","Requirement already satisfied: numpy>=1.9.1 in /usr/local/lib/python3.7/dist-packages (from keras_vggface==0.6) (1.21.5)\n","Requirement already satisfied: h5py in /usr/local/lib/python3.7/dist-packages (from keras_vggface==0.6) (3.1.0)\n","Requirement already satisfied: keras in /usr/local/lib/python3.7/dist-packages (from keras_vggface==0.6) (2.8.0)\n","Requirement already satisfied: pillow in /usr/local/lib/python3.7/dist-packages (from keras_vggface==0.6) (7.1.2)\n","Requirement already satisfied: scipy>=0.14 in /usr/local/lib/python3.7/dist-packages (from keras_vggface==0.6) (1.4.1)\n","Requirement already satisfied: pyyaml in /usr/local/lib/python3.7/dist-packages (from keras_vggface==0.6) (3.13)\n","Requirement already satisfied: cached-property in /usr/local/lib/python3.7/dist-packages (from h5py->keras_vggface==0.6) (1.5.2)\n","Installing collected packages: keras-vggface\n","Successfully installed keras-vggface-0.6\n"]}]}]}